public/bitemp_graph.js

```diff
@@ -77,37 +95,57 @@ var barChart = function() {
       xAxis = d3.svg.axis()
         .scale(xScale)
         .ticks(10)
-        .innerTickSize(-width + axisLabelMargin + margin.left + margin.right)
-        .outerTickSize(0)
-        .orient('bottom')
-        .tickFormat(d3.time.format('%Y-%m-%d'));
+        .tickFormat(d3.time.format('%Y-%m-%d'))
+        .tickSize(8,0)
+        .orient('end');
+
     }
 
     function setupYAxis() {
-      var mindate, maxdate;
+      // minStart: earliest sysStart
+      // maxEnd: latest non-infinty sysEnd
+      // maxStart: max sysStart time
+      var minStart, maxEnd, maxStart;
+
       if (yMin) {
-        mindate = yMin;
+        minStart = yMin;
       }
       else {
-        mindate =
+        minStart =
           moment.min(data.map(function(d){
             return moment(d.content.valStart);
           })).toDate();
       }
+
+      var MIN_MOMENT = moment('1600-01-01T00:00:00');
+
       if (yMax) {
-        maxdate = yMax;
+        maxEnd = yMax;
       }
       else {
-        maxdate =
+        maxEnd =
           moment.max(data.map(function(d){
+            if (d.content.valEnd.startsWith('9999')) {
+              return MIN_MOMENT;
+            }
+            else {
+              return moment(d.content.valEnd);
+            }
             return moment(d.content.valStart);
-          })).add(5, 'y').toDate();
+          })).toDate();
       }
 
-      console.log('ymin='+mindate,' ymax='+maxdate);
+      maxStart =
+        moment.max(data.map(function(d){
+          return moment(d.content.valStart);
+        })).toDate();
+
+      if (maxEnd.getFullYear() === 1600) { // All end dates are infinty
```

Comment:
Yea but I still need a way to tell if all the sysEnd times are infinity. I want to find the max sysEnd time that isn't infinity right?